### PR TITLE
Add support for Nengo 3.1

### PR DESCRIPTION
**Pull request by drasmuss**
Adds a noop SimProbe implementation, which lets us simplify some of the special case code for signals that were probed but not targeted by any ops (since they'll always be targeted by the SimProbe op now).

Also updates an unrelated test involving converting KerasSpiking layers using the NengoDL Converter in graph mode. I have no idea why this suddenly started failing (went back to earlier TensorFlow versions and the problem still occurs, and nothing significant seems to have changed in between the CI build where this last passed and now). But it's such a niche edge case that I didn't spend any more time looking into it.

### CODE REVIEW

File: nengo_dl/tests/test_converter.py [hunk 1]
```diff
@@ -839,7 +839,12 @@ def test_spiking_activation(Simulator, seed, rng):
 
     # note: it will not match for batch_size > 1, because keras-spiking generates
     # a different initial state for each batch element, which is not possible in nengo
-    _test_convert(inp, x, inp_vals=[inp_val], temporal_model=True)
+    model = tf.keras.Model(inp, x)
+    conv = converter.Converter(model, temporal_model=True)
+    assert conv.verify(training=False, inputs=[inp_val])
+    if not pytestconfig.getoption("--graph-mode"):
+        # the keras-spiking gradient implementation doesn't work in graph mode
+        assert conv.verify(training=True, inputs=[inp_val])
```
Comment (hunse) [inline nengo_dl/tests/test_converter.py:847]: Is this a temporary thing (i.e. we should be able to fix this in KerasSpiking for TF 2.4), or is it permanent, or do we not know yet?

**Review by hunse — APPROVED** [review]
I added a test of the `FrozenOrderedSet` backport to get full coverage.

File: nengo_dl/tests/test_converter.py (hunk 1 above)
Comment (hunse) [reply nengo_dl/tests/test_converter.py:847]: Never mind, I read the comment above. I've added an issue to `KerasSpiking` to look into this: nengo/keras-spiking#10.

**Comment by drasmuss** [comment]
Test looks good. I saw the coverage being reported <100% here, but then when I looked on codecov it said 100% coverage so I assumed that was the correct number 🤷.
